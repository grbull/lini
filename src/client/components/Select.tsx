import cn from 'classnames';
import React, { ChangeEvent, ReactElement } from 'react';

interface Props {
  onChange: (event: ChangeEvent<HTMLSelectElement>) => void;
  value?: 'auto' | 'light' | 'dark';
}

export function Select({ onChange, value }: Props): ReactElement {
  return (
    <select
      aria-label="theme-select"
      className={cn('bg-gray-500', 'py-1', 'rounded')}
      defaultValue={value}
      id="theme"
      name="theme"
      onBlur={onChange}
    >
      <option value="auto">Auto (system)</option>

      <option value="dark">Dark</option>
      <option value="light">Light</option>
    </select>
  );
}

Select.defaultProps = { value: 'uto' };
