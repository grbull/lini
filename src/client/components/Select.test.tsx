/**
 * @jest-environment jsdom
 */

import '@testing-library/jest-dom/extend-expect';

import {
  fireEvent,
  render,
  RenderResult,
  waitFor,
} from '@testing-library/react';
import React from 'react';

import { Select } from './Select';

function setup(): RenderResult & {
  onChange: typeof jest.fn;
  select: HTMLSelectElement;
} {
  const onChange = jest.fn();

  const utils = render(<Select onChange={onChange} value="auto" />);

  const select = utils.getByLabelText('theme-select') as HTMLSelectElement;

  return { onChange, select, ...utils };
}

describe('Select Component', () => {
  it('matches the snapshot', () => {
    const { asFragment } = setup();
    expect(asFragment()).toMatchSnapshot();
  });

  it('handles value change', async () => {
    const { onChange, select } = setup();

    fireEvent.change(select, { target: { value: 'light' } });
    fireEvent.focusOut(select);

    await waitFor(() => expect(onChange).toBeCalled());
  });
});
