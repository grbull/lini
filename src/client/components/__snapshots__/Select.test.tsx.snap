// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select Component matches the snapshot 1`] = `
<DocumentFragment>
  <select
    aria-label="theme-select"
    class="bg-gray-500 py-1 rounded"
    id="theme"
    name="theme"
  >
    <option
      selected=""
      value="auto"
    >
      Auto (system)
    </option>
    <option
      value="dark"
    >
      Dark
    </option>
    <option
      value="light"
    >
      Light
    </option>
  </select>
</DocumentFragment>
`;
